### PR TITLE
Refactored run_workload method duplicated in the workloads

### PR DESCRIPTION
**Pull request by vishnuchalla**
### Description
Refactored run_workload method as the same code is duplicated in multiple workloads.
### Fixes
Workloads in which code changes are made and their testing logs
network-perf and scale-perf: http://pastebin.test.redhat.com/1057462
api-load: <a href="https://drive.google.com/file/d/1sYxZlLAyCHB8dnJhBrYYdfAyUJ3eceSd/view?usp=sharing"></a>
cyclic-test: <a href="https://drive.google.com/file/d/1p9j8E4UXomyTZi3LRvs3BFZMNmNDFNpv/view?usp=sharing"></a>
oslat and testpmd: Verified whether the code changes are reachable by adding debug points.

### CODE REVIEW

File: workloads/api-load/run.sh [hunk 1]
```diff
@@ -9,6 +9,8 @@ log "###############################################"
 log "api-load tests: ${TESTS}"
 log "###############################################"
 
+prepare_tests
```
Comment (mukrishn) [inline workloads/api-load/run.sh:12]: @vishnuchalla are these intended ? I think you have to rebase your code and update this PR.

Comment (mukrishn) [reply workloads/api-load/run.sh:12]: okay, that make sense. you moved it out of the function.

**Review by mukrishn — APPROVED** [review]
LGTM

**Review by rsevilla87 — APPROVED** [review]
LGTM

**Review by dry923 — APPROVED** [review]
LGTM!

**Pull request merged**